pkg/controller/controller.go

```diff
@@ -277,7 +280,10 @@ func (c *controller) runController(params ControllerParams) {
 			default:
 				c.getLogger().WithField(fieldConsecutiveErrors, errorRetries).
 					WithError(err).Debug("Controller run failed")
-				c.recordError(err)
+				c.recordError(err, params.HealthReporter)
+				if params.HealthReporter != nil {
+					params.HealthReporter.Degraded(c.name, err)
+				}
```

Comment:
This seems to be duplicated code, `recordError()` already calls `Degraded()`?

Reply:
yup, fixed!